nano/node/rocksdb/rocksdb.cpp

```diff
@@ -111,6 +107,100 @@ void nano::rocksdb_store::open (bool & error_a, boost::filesystem::path const &
+		cf_options.prefix_extractor.reset (rocksdb::NewFixedPrefixTransform (sizeof (nano::root)));
+
+		// Number of files in level 0 which triggers compaction. Size of L0 and L1 should be kept similar as this is the only compaction which is single threaded
+		cf_options.level0_file_num_compaction_trigger = 2;
+
+		// L1 size, compaction is triggered for L0 at this size (2 SST files in L1)
+		cf_options.max_bytes_for_level_base = memtable_size_bytes * 2;
+		cf_options.max_bytes_for_level_multiplier = 10; // Default
+	}
+	else if (cf_name_a == "blocks")
+	{
+		std::shared_ptr<rocksdb::TableFactory> table_factory (rocksdb::NewBlockBasedTableFactory (get_active_table_options (block_cache_size_bytes * 4)));
+		cf_options = get_active_cf_options (table_factory, memtable_size_bytes);
+	}
+	else if (cf_name_a == "confirmation_height")
+	{
+		// Entries will not be deleted in the normal case, so can make memtables a lot bigger
+		std::shared_ptr<rocksdb::TableFactory> table_factory (rocksdb::NewBlockBasedTableFactory (get_active_table_options (block_cache_size_bytes)));
+		cf_options = get_active_cf_options (table_factory, memtable_size_bytes * 2);
+	}
+	else if (cf_name_a == "meta" || cf_name_a == "online_weight" || cf_name_a == "peers")
+	{
+		// Meta - It contains just version key
+		// Online weight - Periodically deleted
+		// Peers - Cleaned periodically, a lot of deletions. This is never read outside of initializing? Keep this small
+		cf_options = get_small_cf_options (small_table_factory);
+	}
+	else if (cf_name_a == "cached_counts")
+	{
+		// Really small (keys are blocks tables, value is uint64_t)
+		cf_options = get_small_cf_options (small_table_factory);
+	}
+	else if (cf_name_a == "pending")
+	{
+		// Pending can have a lot of deletions too
+		std::shared_ptr<rocksdb::TableFactory> table_factory (rocksdb::NewBlockBasedTableFactory (get_active_table_options (block_cache_size_bytes)));
+		cf_options = get_active_cf_options (table_factory, memtable_size_bytes);
+	}
+	else if (cf_name_a == "frontiers")
+	{
+		// Frontiers is only needed during bootstrap for legacy blocks
+		std::shared_ptr<rocksdb::TableFactory> table_factory (rocksdb::NewBlockBasedTableFactory (get_active_table_options (block_cache_size_bytes)));
+		cf_options = get_active_cf_options (table_factory, memtable_size_bytes);
+	}
+	else if (cf_name_a == "accounts")
+	{
+		// Can have deletions from rollbacks
+		std::shared_ptr<rocksdb::TableFactory> table_factory (rocksdb::NewBlockBasedTableFactory (get_active_table_options (block_cache_size_bytes * 2)));
+		cf_options = get_active_cf_options (table_factory, memtable_size_bytes);
+	}
+	else if (cf_name_a == "vote")
+	{
+		// No deletes it seems, only overwrites.
+		std::shared_ptr<rocksdb::TableFactory> table_factory (rocksdb::NewBlockBasedTableFactory (get_active_table_options (block_cache_size_bytes * 2)));
+		cf_options = get_active_cf_options (table_factory, memtable_size_bytes);
+	}
+	else if (cf_name_a == "default")
```

Comment:
Should this be rocksdb::kDefaultColumnFamilyName.c_str ()

Reply:
Yeh makes more sense than hardcoding it.